src/headers/fee.ts

```diff
@@ -0,0 +1,240 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { getVertexHeaderIdBuffer, getVertexHeaderIdFromBuffer, VertexHeaderId } from './types';
+import Header from './base';
+import Network from '../models/network';
+import { IFeeEntry, OutputValueType } from '../types';
+import { intToBytes, unpackToInt } from '../utils/buffer';
+import leb128Util from '../utils/leb128';
+import { MAX_FEE_HEADER_ENTRIES } from '../constants';
+
+/**
+ * FeeHeader represents the fee payment information in a transaction.
+ * It explicitly indicates which tokens and amounts are being used to pay transaction fees.
+ *
+ * @class FeeHeader
+ * @extends {Header}
+ */
+class FeeHeader extends Header {
+  /**
+   * Array of fee entries, each specifying a token and amount used for fee payment
+   */
+  entries: IFeeEntry[];
+
+  /**
+   * Creates an instance of FeeHeader.
+   *
+   * @param {IFeeEntry[]} entries - Array of fee entries
+   * @memberof FeeHeader
+   */
+  constructor(entries: IFeeEntry[]) {
+    super();
+    this.entries = entries;
+  }
+
+  /**
+   * Serialize the fee header fields to a buffer array.
+   * Format:
+   *   [num_entries: 1 byte]
+   *   For each entry:
+   *     [tokenIndex: 1 byte]
+   *     [amount: variable bytes, LEB128 unsigned]
+   *
+   * @param {Buffer[]} array - Array of buffers to push the serialized fields
+   * @memberof FeeHeader
+   * @inner
+   */
+  serializeFields(array: Buffer[]) {
+    // Number of entries
+    array.push(intToBytes(this.entries.length, 1));
+
+    // Serialize each entry
+    for (const entry of this.entries) {
+      // Token Index (1 byte)
+      array.push(intToBytes(entry.tokenIndex, 1));
+
+      // Amount (variable length, LEB128 unsigned)
+      array.push(leb128Util.encodeUnsigned(entry.amount));
```

Comment:
I'm not sure we are using leb128 for serialization of output values. In the core it's using the util method `output_value_to_bytes`, similar to what we are using in the nano header for the action amount `outputValueToBytes`